src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1567 @@
+            SR document instance.
+        institutional_department_name: Union[str, None], optional
+            Name of the department of the person or device that creates the
+            SR document instance.
+        content_creator_name: Union[str, pydicom.valuerep.PersonName, None], optional
+            Name of the person who created the content of this presentation
+            state.
+        content_creator_identification: Union[highdicom.ContentCreatorIdentificationCodeSequence, None], optional
+            Identifying information for the person who created the content of
+            this presentation state.
+        rescale_intercept: Union[int, float, None], optional
+            Intercept used for rescaling pixel values.
+        rescale_slope: Union[int, float, None], optional
+            Slope used for rescaling pixel values.
+        rescale_type: Union[highdicom.RescaleTypeValues, str, None], optional
+            String or enumerated value specifying the units of the output of
+            the Modality LUT or rescale operation.
+        modality_lut: Union[highdicom.ModalityLUT, None], optional
+            Lookup table specifying a pixel rescaling operation to apply to
+            the stored values to give modality values.
+        copy_modality_lut: bool, optional
+            Include elements of the Modality LUT module (including
+            RescaleIntercept, RescaleSlope and ModalityLUTSequence), if any,
+            in the presentation state with values copied from the source images.
+        softcopy_voi_luts: Union[Sequence[highdicom.pr.SoftcopyVOILUT], None], optional
+            One or more pixel value-of-interest operations to applied after the
+            modality LUT and/or rescale operation.
+        copy_voi_lut: bool, optional
+            Include elements of the Softcopy VOI LUT module (including
+            WindowWidth, WindowCenter, and VOILUTSequence), if any, in the
+            presentation state with values copied from the source images.
+        icc_profile: Union[PIL.ImageCms.ImageCmsProfile, None], optional
+            ICC color profile object to include in the presentation state. If
+            none is provided, a standard RGB ("sRGB") profile will be assumed.
+        transfer_syntax_uid: Union[str, highdicom.UID], optional
+            Transfer syntax UID of the presentation state.
+        **kwargs: Any, optional
+            Additional keyword arguments that will be passed to the constructor
+            of `highdicom.base.SOPClass`
+
+        """  # noqa: E501
+        if sop_class_uid not in (
+            GrayscaleSoftcopyPresentationStateStorage,
+            PseudoColorSoftcopyPresentationStateStorage,
+            ColorSoftcopyPresentationStateStorage
+        ):
+            raise ValueError(
+                'The IOD indicated by the provided sop_class_uid is not '
+                'implemented.'
+            )
```

Comment:
Fair point. Removed in 9d03877

Reply:
This inheritance structure never sat that comfortably with me to be honest. This would seem to be a good case for composition over inheritance. This was the best I could come up with but it's not perfect, the base classes and the derived classes remain quite tightly coupled

Reply:
Yeah...we could consider using setter functions that set attributes of a module or macro on an existing dataset without inheritance. However, in this particular case, I don't think inheritance is necessarily a bad idea, because each of the derived classes conceptually is a different "type" of [Presentation State](https://dicom.nema.org/medical/dicom/current/output/chtml/part03/chapter_A.html#sect_A.1.2.11) and they all have a lot of attributes in common.